lib/assert.js

```diff
@@ -169,11 +169,11 @@ x.snapshot = function (tree, match, snapshotStateGetter) {
 	var result = toMatchSnapshot.call(context, tree);
 
 	var messagePrefix = 'Please check your code or --update-snapshots\n\n';
-	var message = typeof result.message === 'function' && (messagePrefix + result.message());
+	var message = optionalMessage || (typeof result.message === 'function' && (messagePrefix + result.message()));
```

Comment:
The message argument should be in addition to the default output, not replace it. Probably best to prepend before `messagePrefix`.